p8s.go

```diff
@@ -54,14 +55,22 @@ func addRequest(labels map[string]string, logline map[string]interface{}) {
 		if len(uniqueHostnameMap) < maxUniqueHostnames {
 			uniqueHostnameMap[labels["hostname"]] = struct{}{}
 		} else {
-			// Use hard-coded hostname so wildcard domains don't make cardinality explode.
+			// Use hard-coded hostname so wildcard domains don't make
+			// cardinality explode.
 			labels["hostname"] = "max-hostnames-reached"
 		}
 	}
 
 	bytes := getBytes(logline)
 
-	requestsTotal.With(labels).Inc()
+	if isGeoHashing {
+		labels = reduceGeoHashLabels(labels)
+		requestsTotal.With(labels).Inc()
+		labels = scrubGeoHashLabels(labels)
```

Comment:
Is there a way we can accidentally have the `lat` and `lon` labels become metric labels because a log included these values but geo hashing wasn't enabled?

Reply:
At a glance, it looks like the labels are set explicitly. For e.g. `SetupWithGeoHash` sets up the `geo_hash` label. I don't see this happening but would love to be corrected.

Reply:
This has been revised due to how many different paths the code takes to make the labels match.